src/variable.h

```diff
@@ -350,6 +352,74 @@ Variable makeVariable(Tag tag, const Dimensions &dimensions, Args &&... args) {
   }
 }
 
+
+inline void incCoord(const std::vector<size_t> limits, std::vector<size_t>& coords) {
+  auto i = coords.size();
+  bool overflow = true;
+  while (i --> 0) {
+    auto buf = coords[i] + 1;
+    if (buf >= limits[i])
+      coords[i] = 0;
+    else {
+      coords[i] = buf;
+      overflow = false;
+      break;
+    }
+  }
+  if(overflow)
+    throw std::logic_error("Can't increment coordinates: overflow");
+}
+
+
+template  <typename T, typename SZ_TP>
+Variable makeVariable(Tag tag,
+                      const Dimensions &dimensions,
+                      const std::vector<SZ_TP> &stridesInBytes,
+                      T* ptr) {
+  auto ndims = dimensions.ndim();
+  if (ndims == 0) // empty dataset
+    return makeVariable<T>(tag, dimensions);
+
+  std::vector<SZ_TP> varStrides(ndims, 1), strides;
+  for(auto&& strd: stridesInBytes)
+    strides.emplace_back(strd / sizeof(T));
+
+  bool sameStrides{*strides.rbegin() == 1};
+  auto i = varStrides.size() - 1;
+  while(i --> 0) {
+    varStrides[i] = varStrides[i + 1] * dimensions.size(i + 1);
+    if (varStrides[i] != strides[i] && sameStrides)
+      sameStrides = false;
+  }
+
+  if (sameStrides) { //memory is alligned c-style and dense
+    return Variable(tag, defaultUnit(tag), std::move(dimensions),
+                    Vector<underlying_type_t<T>>(ptr, ptr + dimensions.volume()));
+
+  } else {
+    // Try to find blocks to copy
+    auto index = strides.size() - 1;
+    while (strides[index] == varStrides[index]) --index;
+    ++index;
+    auto blockSz = index < strides.size() ? strides[index] * dimensions.size(index) : 1;
+
+    auto res = makeVariable<T>(tag, dimensions);
+    std::vector<size_t> dsz(ndims);
+    for (size_t i = 0; i < index; ++i)
+      dsz[i] = dimensions.size(i);
+    std::vector<size_t> coords(ndims, 0);
+    res.template span<T>()[0] = ptr[0];
```

Comment:
Why is element 0 assigned separately?

Reply:
Because the `incCoord` would throw the exception if the overflow occurs, so couldn't make it at the end of the loop, or should put if statement inside the loop, or better (like it is done) to treat the first value before the loop and make `incCoord` at the beginning of the loop, then no overflow occurs.